sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala

```diff
@@ -132,3 +132,32 @@ case class CheckOverflow(
 
   override def sql: String = child.sql
 }
+
+case class HasOverflow(
+    child: Expression,
+    inputType: DecimalType) extends UnaryExpression {
+
+  override def dataType: DataType = BooleanType
+
+  override def nullable: Boolean = true
```

Comment:
Can we override `nullable` with `false` as `doGenCode()` does?

Reply:
Since the child can be nullable, the input value can be null.   Making nullable to false in that case will not work, as it may result in npe.  We can change the doGenCode() here to make the check for null for that, but since  the nullSafeCodeGen in UnaryExpression already takes care of the if nullable checks,  it seems there is no need to add if null checks here.